### DISCUSSION TITLE
ACSelect set element by index value

### DISCUSSION THREAD

**Discussion by Koxx3** [Hieromon/AutoConnect#330] [Q&A] [answered]
Hello,
currently, I set ACSelect element value by using a dirty if/else :

![image](https://user-images.githubusercontent.com/11454444/111138944-d3c2d000-8580-11eb-8b75-20bdad0d1af0.png)

Did I miss something ?
Could it be possible to add a function to set the value from index ?

Getting the selected index is much easier and work fine ;)

![image](https://user-images.githubusercontent.com/11454444/111139179-1a182f00-8581-11eb-9c0b-f7d5f7b3d837.png)

Nota : yes, I have to use my own store/load function in a more larger settings environment (BLE, TFT/keys ...)

**Comment by Hieromon** [comment]
@Koxx3 
>Could it be possible to add a function to set the value from index ?

What does the index point to? Do you want to add a string as the nth-option that can be selected as a SELECT element?
That is, does that mean you want to control the order of selectable options string when they appear on a web page as HTML elements?
If so, you cannot. ACSelect has the values as `std::vector<String>`. That it to each element linearly, and stored while maintaining the order.

**NOTE**
This topic has converted to the [discussions](https://github.com/Hieromon/AutoConnect/discussions).

**Reply by Koxx3** [reply]
Let's take an example :

`ACSelect(ACE_SETTINGS_Default_eco_mode_at_startup, {"Max","Medium","None"}, "Startup eco mode", 2);`

I'd like to be able to do : 

`        aux.setElementValueByIndex("ACE_SETTINGS_Default_eco_mode_at_startup", 1);`

... to select element number 1, aka "Medium".

**Reply by Hieromon** [reply]
**EDIT**
 This reply has changed to answer.

~~The substance of the element value of ACSelect is stored in the *sequence container* created by [**std::vector**](https://en.cppreference.com/w/cpp/container/vector). Therefore, it cannot store the value by specifying an index that directly indexes the element value.
Also, the [insertion of std::vector](https://en.cppreference.com/w/cpp/container/vector/insert) can be positioned using an iterator, but it is a relative position, not the absolute coordinates indicated by the so-called index.
(e.g. If the container has 1 elements currently and you try to insert it 3rd (as index), the result will be undefined. (which will probably cause a segfault)
Therefore, your requirement does not conform to the data structure of ACSelect.~~

**Comment by Hieromon** [comment, marked as answer]
The substance of the element value of ACSelect is stored in the sequence container created by [**std::vector**](https://en.cppreference.com/w/cpp/container/vector). Therefore, it cannot store the value by specifying an index that directly indexes the element value.
Also, the [insertion of std::vector](https://en.cppreference.com/w/cpp/container/vector/insert) can be positioned using an iterator, but it is a relative position, not the absolute coordinates indicated by the so-called index.
(e.g. If the container has 1 elements currently and you try to insert it 3rd (as index), the result will be undefined. (which will probably cause a segfault)
Therefore, your requirement does not conform to the data structure of ACSelect.
